test/unit/devicepreferences.test.ts

```diff
@@ -0,0 +1,87 @@
+
+	afterEach(() => {
+		axios.request.mockReset()
+	})
+
+	test('list', async () => {
+		axios.request.mockImplementationOnce(() => Promise.resolve({ status: 200, data: preferencesList }))
+
+		const response = await client.devicePreferences.list()
+
+		expect(axios.request).toHaveBeenCalledTimes(1)
+		expect(axios.request).toHaveBeenCalledWith(buildRequest('devicepreferences', {}))
+		expect(response).toBe(preferencesList.items)
+	})
+
+	test('list with namespace', async () => {
+		axios.request.mockImplementationOnce(() => Promise.resolve({ status: 200, data: preferencesList }))
+
+		const response = await client.devicePreferences.list('my-namespace')
+
+		expect(axios.request).toHaveBeenCalledTimes(1)
+		expect(axios.request).toHaveBeenCalledWith(buildRequest('devicepreferences', { namespace: 'my-namespace'}))
+		expect(response).toBe(preferencesList.items)
+	})
+
+	test('get minimal', async () => {
+		axios.request.mockImplementationOnce(() => Promise.resolve({ status: 200, data: minimalIntegerPreference }))
+
+		const response = await client.devicePreferences.get('my-id')
+
+		expect(axios.request).toHaveBeenCalledTimes(1)
+		expect(axios.request).toHaveBeenCalledWith(buildRequest('devicepreferences/my-id'))
+		expect(response).toBe(minimalIntegerPreference)
+	})
+
+	test('get', async () => {
+		axios.request.mockImplementationOnce(() => Promise.resolve({ status: 200, data: integerPreference }))
+
+		const response = await client.devicePreferences.get('my-id')
+
+		expect(axios.request).toHaveBeenCalledTimes(1)
+		expect(axios.request).toHaveBeenCalledWith(buildRequest('devicepreferences/my-id'))
+		expect(response).toBe(integerPreference)
+	})
+
+	test('create', async () => {
+		axios.request.mockImplementationOnce(() => Promise.resolve({ status: 200, data: integerPreference }))
+
+		const response = await client.devicePreferences.create(integerPreferenceCreate)
+
+		expect(axios.request).toHaveBeenCalledTimes(1)
+		expect(axios.request).toHaveBeenCalledWith(buildRequest('devicepreferences', undefined, integerPreferenceCreate, 'post'))
+		expect(response).toBe(integerPreference)
+	})
+
+	test('update', async () => {
+		axios.request.mockImplementationOnce(() => Promise.resolve({ status: 200, data: integerPreference }))
+
+		const response = await client.devicePreferences.update('my-id', minimalIntegerPreference)
+
+		expect(axios.request).toHaveBeenCalledTimes(1)
+		expect(axios.request).toHaveBeenCalledWith(buildRequest('devicepreferences/my-id', undefined, minimalIntegerPreference, 'put'))
+		expect(response).toBe(integerPreference)
+	})
+
+	test('delete', async () => {
+		axios.request.mockImplementationOnce(() => Promise.resolve({ status: 200, data: {} }))
+
+		const response = await client.devicePreferences.delete('my-id')
+
+		expect(axios.request).toHaveBeenCalledTimes(1)
+		expect(axios.request).toHaveBeenCalledWith(buildRequest('devicepreferences/my-id', undefined, undefined, 'delete'))
+		expect(response).toBe(SuccessStatusValue)
```

Comment:
We are missing failure cases here and in `endpoint-client` tests. It would be a good idea to add some coverage on common failure behavior in methods other than jest GET.